client/client.go

```diff
@@ -1124,6 +1134,54 @@ func (cli Client) PostInvestigatorStatus(iid float64, newstatus string) (err err
 	return
 }
 
+// PostInvestigatorAPIKey is used to either enable or disable API key based access
+// to the MIG API for an investigator. API key based access to the API can be used in
+// place of X-PGPAUTHORIZATION API authentication.
+//
+// If an API key is being set, the returned investigator APIKey value will contain the
+// assigned key. newstatus should be set to either 'active' or 'disabled'. If a key already
+// exists for an investigator, calling this with a status of 'active' will cause the existing
+// key to be replaced.
+func (cli Client) PostInvestigatorAPIKey(iid float64, newstatus string) (inv mig.Investigator, err error) {
```

Comment:
I wonder if posting an api key is something we want, versus `GET /investigator/apikey` that would return a randomly generated one. Thoughts?